DatadogCore/Sources/Core/Context/CarrierInfoPublisher.swift

```diff
@@ -64,6 +64,12 @@ extension CarrierInfo {
 
 // MARK: - iOS 11
 
+// Xcode 15 ships with Swift 5.9 and drops support for iOS 11 deployment target.
+// The conditional check ensures that iOS 11 components are only compiled with Xcode 14, where iOS 11
+// remains a valid deployment target. Since this deployment target is no longer available in Xcode 15,
+// compiling the following code in Xcode 15 would result in compiler warnings (due to deprecation in some used APIs),
+// violating our 0 warnings policy. It can be entirely removed when Xcode 14 is dropped.
+#if swift(<5.9)
```

Comment:
/question can we use `@available(iOS, deprecated: 12, obsoleted: 13, message: "something") instead or does it show a warning still?

Reply:
Actually, it was even simpler. What worked is: `@available(iOS, deprecated: 12)`. No warnings with 11.0, 12.0 or 13.0 deployment target (both in Xcode 14.x and 15.x). ✅ changed. Thanks, a good call 👍.